--- CMakeLists.txt.orig	2025-12-17 01:21:47
+++ CMakeLists.txt	2025-12-17 01:21:47
@@ -238,14 +238,7 @@
 #
 # Main target library
 #
-if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
-  set(RE_LIBRARIES ${_VCPKG_INSTALLED_DIR}/x64-linux/lib/libre.so)
-else()
-  set(RE_LIBRARIES ${_VCPKG_INSTALLED_DIR}/arm64-linux/lib/libre.so)
-endif(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
-
-set(LINKLIBS ${RE_LIBRARIES} Threads::Threads)
-
+set(LINKLIBS re Threads::Threads)
 if(ZLIB_FOUND)
   list(APPEND LINKLIBS ZLIB::ZLIB)
 endif()
